src/main/java/com/bandwidth/sdk/BandwidthClient.java

```diff
@@ -196,6 +200,27 @@ public void setCredentials(final String userId, final String apiToken, final Str
         usersUri = String.format(BandwidthConstants.USERS_URI_PATH, userId);
         this.token = apiToken;
         this.secret = apiSecret;
+
+        validateCredentials();
+    }
+
+    /**
+     * Validate if the credentials are set and has access to catapult
+     */
+    private static void validateCredentials() {
+        try {
+            Account.get().getAccountInfo();
+        } catch (Exception e) {
+            if (e instanceof AppPlatformException) {
+                AppPlatformException appEx = (AppPlatformException) e;
+
+                if (appEx.getStatus() == 401){
```

Comment:
If 400 is returned, it means that the credentials are invalid not missing, isn't it?